### PR TITLE
TDL-19413: Products stream will not be supported in new version for certain PIMs

### PR DESCRIPTION
**Pull request by hpatel41**
# Description of change
TDL-19413: Products stream will not be supported in new version for certain PIMs
- Updated the code to use Old API Verison (2021-01) for the Products stream.
- **NOTE:** In the [API Changelog](https://docs.rechargepayments.com/changelog/2021-11-api-release-notes/#2-products-and-plans) it is mentioned that to get Products in New Version: 2021-11 we can use `/plans`, but we have found major field changes ([Tap-Recharge_ Products vs Plans.pdf](https://github.com/singer-io/tap-recharge/files/8907490/Tap-Recharge_.Products.vs.Plans.pdf)) for Plans endpoint thus, we will use Old Verison for this endpoint.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tests/unittests/test_products_stream_verison.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    """
+
+    recharge_client = RechargeClient('test_access_token')
+
+    def test_products_stream_version(self, mocked_request):
+        """
+            Test case to verify we are calling stream 'Products' with API Version: '2021-01'
+        """
+        # mock request and return value with data key
+        mocked_request.return_value = Mockresponse({'products': [{'key': 'value'}]})
```
Comment (savan-chovatiya) [inline tests/unittests/test_products_stream_verison.py:29]: Move this return_value at the decorator level.

File: tests/unittests/test_products_stream_verison.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+        # verify we using API Version: '2021-01'
+        self.assertEqual(request_headers.get('X-Recharge-Version'), '2021-01')
+
+    def test_non_products_stream_version(self, mocked_request):
+        """
+            Test case to verify we are calling stream other than 'Products' with API Version: '2021-11'
+        """
+        # mock request and return value with data key
+        mocked_request.return_value = Mockresponse({'collections': [{'key': 'value'}]})
```
Comment (savan-chovatiya) [inline tests/unittests/test_products_stream_verison.py:49]: Move this return_value at the decorator level.

File: tests/unittests/test_products_stream_verison.py (hunk 1 above)
Comment (harshpatel4crest) [reply tests/unittests/test_products_stream_verison.py]: Updated the unittest.

File: tests/unittests/test_products_stream_verison.py (hunk 2 above)
Comment (harshpatel4crest) [reply tests/unittests/test_products_stream_verison.py]: Updated the unittest.

File: tap_recharge/client.py [hunk 3]
```diff
@@ -190,6 +190,12 @@ def request(self, method, path=None, url=None, **kwargs): # pylint: disable=too-
         if path == 'collections':
             kwargs['headers']['X-Recharge-Version'] = '2021-11'
 
+        # In API Version: 2021-11, Products endpoint is only available for merchants doing a custom integration
+        # with a PIM (Product Information Management) system that isn’t Shopify or BigCommerce.
```
[the remark this answers is not in the document]
Comment (harshpatel4crest) [reply tap_recharge/client.py]: Updated comment.

File: tap_recharge/client.py [hunk 4]
```diff
@@ -190,6 +190,12 @@ def request(self, method, path=None, url=None, **kwargs): # pylint: disable=too-
         if path == 'collections':
```
Comment (dbshah1212) [inline tap_recharge/client.py:190]: How this PR is different then https://github.com/singer-io/tap-recharge/pull/32/files
Can we merge everything in one?

Comment (harshpatel4crest) [reply tap_recharge/client.py:190]: This, PR contains all the changes related to the Products stream. And PR #32 is general missing fields PR for all the streams.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

**Review by RushiT0122 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**